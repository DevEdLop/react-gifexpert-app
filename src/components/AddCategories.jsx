import { useState } from "react"
import PropTypes from 'prop-types'
export const AddCategories = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (e) => {
        setInputValue(e.target.value)
    }
    const onSubmit = (e) => {
        e.preventDefault()
        if (inputValue.trim().length <= 1) return;
        // setCategories(categories => [inputValue, ...categories])
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    return (
        <form
            aria-label="form"
            onSubmit={onSubmit}>
            <input
                aria-label="input-gif"
                type={'text'}
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}


AddCategories.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}

