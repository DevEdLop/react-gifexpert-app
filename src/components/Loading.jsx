import React from 'react'



const Loading = ({ message }) => {
    return (
        <div className="card-grid">
            <div className="lds-ellipsis">
                <div></div>
                <div></div>
                <div></div>
                <div></div>
                <h5 style={{margin: '4px'}}>

                    {message}

                </h5>
            </div>
        </div>
    )
}

export default Loading
