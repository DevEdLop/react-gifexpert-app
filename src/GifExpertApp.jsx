import { useState } from "react"
import { AddCategories, GifGrid } from "./components"

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball'])

    const onAddCategories = (newCategory) => {
        if (categories.find(item => item.toLowerCase() === newCategory.toLowerCase())) return alert('Ya existe esta wea');
        setCategories([newCategory, ...categories])

    }

    // console.log(categories)
    return (

        <>
            <h1>GifExpertApp</h1>

            <AddCategories
                onNewCategory={onAddCategories}
            />
            {
                categories.map((item) => (
                    <GifGrid
                        key={item}
                        category={item}
                    />
                ))
            }
        </>
    )
}

export default GifExpertApp