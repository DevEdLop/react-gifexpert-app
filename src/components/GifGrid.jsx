import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import Loading from "./Loading";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category)

    return (
        <>
            <h3>{category}</h3>
            {isLoading && (
                <Loading
                    message={'Generando...'}
                />
            )}
            <div className="card-grid">

                {
                    images.map((image) => (

                        <GifItem
                            key={image.id}
                            {...image}
                        />)
                    )
                }
            </div>



        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}