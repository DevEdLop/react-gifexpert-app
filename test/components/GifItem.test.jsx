import { render, screen } from "@testing-library/react"
import { GifItem } from '../../src/components'
describe('Prueba a <GrifItem/>', () => {
    const title = 'Dragon Ball';
    const url = 'https://giphy.com/gifs/TOEIAnimationUK-goku-dragon-ball-super-19JSJ5ucu91R5D7a3w';
    test('debe hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()

    })

    test('debe de mostrar la imagen con el URL y ALT indicado', () => {

        render(<GifItem title={title} url={url} />);

        const { src, alt } = screen.getByRole('img')

        expect(src).toBe(url)
        expect(alt).toBe('yatusabe')
        // screen.debug()

    })

    test('debe de mostrar el titulo en el componente', () => {

        render(<GifItem title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy()
    })

})
