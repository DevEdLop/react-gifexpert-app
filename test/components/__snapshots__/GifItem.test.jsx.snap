// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba a <GrifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="yatusabe"
      src="https://giphy.com/gifs/TOEIAnimationUK-goku-dragon-ball-super-19JSJ5ucu91R5D7a3w"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
